python/lsst/daf/butler/core/storedFileInfo.py

```diff
@@ -19,87 +19,51 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-__all__ = ("StoredFileInfo", )
+__all__ = ("StoredFileInfo", "StoredDatastoreItemInfo")
 
 import inspect
+from dataclasses import dataclass
+from typing import Optional
 
 from .formatter import Formatter
-from .utils import slotValuesAreEqual
 from .storageClass import StorageClass
 
 
-class StoredFileInfo:
-    """Information associated with a stored file in a Datastore.
+class StoredDatastoreItemInfo:
+    """Internal information associated with a stored dataset in a `Datastore`.
 
-    Parameters
-    ----------
-    formatter : `str` or `Formatter` or `Formatter` class.
-        Full name of formatter to use to read this Dataset or a `Formatter`
-        instance or class.
-    path : `str`
-        Path to Dataset, relative to `Datastore` root.
-    storageClass : `StorageClass`
-        `StorageClass` used when writing the file. This can differ from that
-        used to read the file if a component is being requested from
-        a concrete composite.
-
-    See Also
-    --------
-    StorageInfo
+    This is an empty base class.  Datastore implementations are expected to
+    write their own subclasses.
     """
+    pass
```

Comment:
Might want to do `__slots__ = ()` here so derived class that have a reason to be compact can be.